libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -9,29 +9,119 @@ import {
+                    case 'signin/verifyState':
+                        return await this.onTeamsSigninVerifyState(context, context.activity.value);
+
+                    case 'fileConsent/invoke':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsFileConsent(context, context.activity.value));
+
+                    case 'actionableMessage/executeAction':
+                        await this.onTeamsO365ConnectorCardAction(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse();
+
+                    case 'composeExtension/queryLink':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsAppBasedLinkQuery(context, context.activity.value));
+
+                    case 'composeExtension/query':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsMessagingExtensionQuery(context, context.activity.value));
+
+                    case 'composeExtension/selectItem':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsMessagingExtensionSelectItem(context, context.activity.value));
+
+                    case 'composeExtension/submitAction':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsMessagingExtensionSubmitActionDispatch(context, context.activity.value));
+
+                    case 'composeExtension/fetchTask':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsMessagingExtensionFetchTask(context, context.activity.value));
+
+                    case 'composeExtension/querySettingUrl':
+                        return TeamsActivityHandler.createInvokeResponse(await this.onTeamsMessagingExtensionConfigurationQuerySettingUrl(context, context.activity.value));
+
+                    case 'composeExtension/setting':
+                        await this.onTeamsMessagingExtensionQuerySetting(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse();
+
+                    case 'composeExtension/onCardButtonClicked':
+                        await this.onTeamsMessagingExtensionCardButtonClicked(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse();
+
+                    case 'task/fetch':
+                        const fetchResponse = await this.onTeamsTaskModuleFetch(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse(fetchResponse);
+
+                    case 'task/submit':
+                        const submitResponse = await this.onTeamsTaskModuleSubmit(context, context.activity.value);
+                        return TeamsActivityHandler.createInvokeResponse(submitResponse);
+
+                    default:
+                        throw new Error('NotImplemented');
```

Comment:
>NotImplemented [](start = 41, length = 14)

Was this going to be a 400 or 404?

Reply:
I just pushed another commit that changes this to 400